### ISSUE TITLE
CI fails test 1661 bufref leak on Linux/macOS torture

### ISSUE THREAD

**Issue by vszakats** [curl/curl#14908]
### I did this

While working on PRs, I notice these CI failure that started coming up fairly consistently around yesterday:
```
FAIL 1661: 'bufref unit tests' unittest, bufref
```
https://github.com/curl/curl/actions/runs/10861785973/job/30144082984?pr=14859#step:15:4424 (macOS)
https://github.com/curl/curl/actions/runs/10861785979/job/30144081788?pr=14859#step:8:4090 (Linux)

```
test 1661...[bufref unit tests]
 2 functions to make fail
** MEMORY FAILURE
Leak detected: memory still allocated: 13 bytes
At 55a2378fd348, there's 13 bytes.
 allocated by unit1661.c:70
LIMIT strdup.c:114 malloc reached memlimit
 1661: torture FAILED: function number 1 in test.
 invoke with "-t1" to repeat this single case.
```

### I expected the following

No new CI failure.

### curl/libcurl version

master

### operating system

Linux, macOS.

**Comment by vszakats** [comment]
Added test details.

**Comment by bagder** [comment]
It does not reproduce for me locally on Linux.

Also, for me it says `1 functions to make fail`, while the failing logs say **2**. 

I can only spot a single function that can fail: a malloc().

**Comment by vszakats** [comment]
Sorry about the noise, It might be a false alarm, and caused by something in one
of the open PRs (suspect: #14772). Though I can't recall seeing it there earlier.

https://github.com/curl/curl/compare/8a33741a6897bf951841aaafe3e004bc2b75c860...0ed50e468c280ffe86f0f2d426edb1dde1d5b949?expand=1

https://github.com/curl/curl/compare/1694ae0089cbc93033accbd6ed44fa18c9238146...863938f5aa676ab534da32b15c8823f8633e0ff9?expand=1

https://github.com/curl/curl/actions/runs/10854786731/job/30125980323 green
https://github.com/curl/curl/actions/runs/10856810350/job/30132186464 red

Closing for now.

**Comment by vszakats** [comment]
I suspect this is a fallout from the test bundle effort (#14772), which remained hidden because of another bug that failed to steer runtests to the bundle exe for autotools builds, making these tests silently skipped. (Hence the lower test run times, now all marked with FIXME since I discovered this). The torture tests are all autotools builds. I did not push a new CI run after fixing the autotools issue, so did not see them fail. I saw them first when rebasing on master and incorporating the full patch into the Azure migration PR #14859. (and missed to make the connection right away.)

This seems like a fun fallout to investigate for #14772! Now at 5 sub-TODO to fix there, though 4 of those are related to bundling _servers_, which I may be splitting off to a separate PR.
